source/constructs/api/data_source/service.py

```diff
@@ -361,13 +381,338 @@
+                    else:
+                        response = glue.create_connection(
+                            ConnectionInput={
+                                'Name': glue_connection_name,
+                                'Description': glue_connection_name,
+                                'ConnectionType': 'JDBC',
+                                'ConnectionProperties': {
+                                    'SECRET_ID': jdbc.secret,
+                                    'JDBC_CONNECTION_URL': jdbc.jdbc_connection_url,
+                                    'JDBC_ENFORCE_SSL': 'false',
+                                },
+
+                                'PhysicalConnectionRequirements': {
+                                    'SubnetId': jdbc.network_subnet_id,
+                                    'AvailabilityZone': jdbc.network_availability_zone,
+                                    'SecurityGroupIdList': [jdbc.network_sg_id]
+                                }
+                            }
+                        )
+                        logger.info(response)
+                try:
+                    glue.get_database(Name=glue_database_name)
+                except Exception as e:
+                    logger.info("sync_rds_connection get_database error and create:")
+                    logger.info(str(e))
+                    response = glue.create_database(DatabaseInput={'Name': glue_database_name})
+                    logger.info(response)
+                lakeformation = boto3.client('lakeformation',
+                                             aws_access_key_id=credentials['AccessKeyId'],
+                                             aws_secret_access_key=credentials['SecretAccessKey'],
+                                             aws_session_token=credentials['SessionToken'],
+                                             region_name=jdbc.region)
+                """ :type : pyboto3.lakeformation """
+                # retry for grant permissions
+                num_retries = GRANT_PERMISSIONS_RETRIES
+                while num_retries > 0:
+                    try:
+                        response = lakeformation.grant_permissions(
+                            Principal={
+                                'DataLakePrincipalIdentifier': f"{crawler_role_arn}"
+                            },
+                            Resource={
+                                'Database': {
+                                    'Name': glue_database_name
+                                }
+                            },
+                            Permissions=['ALL'],
+                            PermissionsWithGrantOption=['ALL']
+                        )
+                    except Exception as e:
+                        sleep(SLEEP_MIN_TIME)
+                        num_retries -= 1
+                    else:
+                        break
+                else:
+                    raise Exception('UNCONNECTED')
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using `Exception` and `BaseException` can make your code prone to errors and difficult to maintain. Instead, we recommend using one of the [Built-in Exceptions](https://docs.python.org/3/library/exceptions.html) or creating a custom exception class that is derived from `Exception` or one of its subclasses.